HDWalletKit_Tests/CryptoTests.swift

```diff
@@ -0,0 +1,99 @@
+        XCTAssertEqual(encrypted.toHexString(), "06b3dfaec148fb1bb2b066f10ec285e7c9bf402ab32aa78a5d38e34566810cd2")
+    }
+    
+    func testPrivateKeySign() {
+        let signer = EIP155Signer()
+        
+        let rawTransaction1 = RawTransaction(
+            value: Wei("10000000000000000")!,
+            to: "0x91c79f31De5208fadCbF83f0a7B0A9b6d8aBA90F",
+            gasPrice: 99000000000,
+            gasLimit: 21000,
+            nonce: 2
+        )
+        
+        XCTAssertEqual(
+            try! signer.hash(rawTransaction: rawTransaction1).toHexString(),
+            "de6ed032e8f09adb557f6a0ebc16ed52d6a75e0644a77a236aa1cfffa7746e9a"
+        )
+        
+        let rawTransaction2 = RawTransaction(
+            value: Wei("10000000000000000")!,
+            to: "0x88b44BC83add758A3642130619D61682282850Df",
+            gasPrice: 99000000000,
+            gasLimit: 200000,
+            nonce: 4
+        )
+        
+        XCTAssertEqual(
+            try! signer.hash(rawTransaction: rawTransaction2).toHexString(),
+            "b148272b2a985365e08abb17a85ca5e171169978f3b55e6852a035f83b9f3aa5"
+        )
+        
+        let rawTransaction3 = RawTransaction(
+            value: Wei("20000000000000000")!,
+            to: "0x72AAb5461F9bE958E1c375285CC2aA7De89D02A1",
+            gasPrice: 99000000000,
+            gasLimit: 21000,
+            nonce: 25
+        )
+        
+        XCTAssertEqual(
+            try! signer.hash(rawTransaction: rawTransaction3).toHexString(),
+            "280e29f030cfa256b4298a2b834a4add92b37f159b3cce1110e1ff9f7514f9fe"
+        )
+    }
+    
+    func testGeneratingRSV() {
+        let signiture = Data(hex: "28ef61340bd939bc2195fe537567866003e1a15d3c71ff63e1590620aa63627667cbe9d8997f761aecb703304b3800ccf555c9f3dc64214b297fb1966a3b6d8300")
+        let signer = EIP155Signer()
+        let (r, s, v) = signer.calculateRSV(signiture: signiture)
+        XCTAssertEqual(r, BInt("18515461264373351373200002665853028612451056578545711640558177340181847433846")!)
+        XCTAssertEqual(s, BInt("46948507304638947509940763649030358759909902576025900602547168820602576006531")!)
+        XCTAssertEqual(v, BInt(37))
+        let restoredSignature = signer.calculateSignature(r: r, s: s, v: v)
+        XCTAssertEqual(signiture, restoredSignature)
+    }
+    
+    func testRestoringSignatureSignedWithOldScheme() {
+        let v = 27
+        let r = "75119860711638973245538703589762310947594328712729260330312782656531560398776"
+        let s = "51392727032514077370236468627319183981033698696331563950328005524752791633785"
+        let signer = EIP155Signer()
+        let signature = signer.calculateSignature(r: BInt(r)!, s: BInt(s)!, v: BInt(v))
+        XCTAssertEqual(signature.toHexString(), "a614559de76862bb1dbf8a969d8979e5bf21b72c51c96b27b3d247b728ebffb8719f40b018940ffd0880285d2196cdd31a710bf7cdda60c77632743d687dff7900")
+    }
+    
+    func testRestoringSignatureSignedWithOldScheme2() {
```

Comment:
@impul join this and previous tests in one

Reply:
done